docker/service/src/main/java/uk/nhs/adaptors/scr/models/EventListQueryResponseParser.java

```diff
@@ -27,34 +26,46 @@
         "//accessControlAssertion[descendant::code[text()='Store'] and descendant::type[text()='SCR']]/permission";
     private static final String LATEST_SCR_ID_XPATH = "(//*[local-name()='event']/*[local-name()='eventID'])[1]";
     private static final String ERROR_REASON_CODE_XPATH = "//justifyingDetectedIssueEvent/code";
+    private static final String ERROR_REASON_CODE_CODE_XPATH = "./@code";
+    private static final List<String> CASE_NOT_FOUND_ERROR_CODES = asList("210", "30312");
+    private static final String SUCCESS_RESPONSE_WRAPPER_XPATH = "//QUPC_IN200000SM04";
 
     private final XmlUtils xmlUtils;
 
     @SneakyThrows
     public EventListQueryResponse parseXml(Document soapEnvelope) {
         EventListQueryResponse response = new EventListQueryResponse();
-        response.setViewPermission(getPermissionValue(soapEnvelope, VIEW_PERMISSION_XPATH));
-        response.setStorePermission(getPermissionValue(soapEnvelope, STORE_PERMISSION_XPATH));
-        response.setLatestScrId(xmlUtils.getNodeAttributeValue(soapEnvelope, LATEST_SCR_ID_XPATH, "root"));
+        if (isSuccessResponse(soapEnvelope) || isCaseNotFound(getErrors(soapEnvelope))) {
+            response.setViewPermission(getPermissionValue(soapEnvelope, VIEW_PERMISSION_XPATH));
+            response.setStorePermission(getPermissionValue(soapEnvelope, STORE_PERMISSION_XPATH));
+            response.setLatestScrId(xmlUtils.getNodeAttributeValue(soapEnvelope, LATEST_SCR_ID_XPATH, "root"));
+        } else {
+            response.setViewPermission(NO);
+            response.setStorePermission(NO);
+        }
 
         return response;
     }
 
     private AcsPermission getPermissionValue(Document soapEnvelope, String xPath) {
-        boolean isErrorSet = isErrorSet(soapEnvelope);
-        if (isErrorSet) {
-            return NO;
-        } else {
-            String nodeText = xmlUtils.getNodeText(soapEnvelope, xPath);
-            return nodeText != null ? AcsPermission.fromValue(nodeText) : ASK;
-        }
+        String nodeText = xmlUtils.getNodeText(soapEnvelope, xPath);
+        return nodeText != null ? AcsPermission.fromValue(nodeText) : ASK;
     }
 
     @SneakyThrows
-    private static boolean isErrorSet(Document document) {
-        XPathExpression xPathExpression = XPathFactory.newInstance().newXPath().compile(ERROR_REASON_CODE_XPATH);
-        NodeList nodeList = ((NodeList) xPathExpression.evaluate(document, NODESET));
+    private List<String> getErrors(Document document) {
+        return xmlUtils.getNodesByXPath(document, ERROR_REASON_CODE_XPATH)
+            .stream()
+            .map(it -> xmlUtils.getValueByXPath(it, ERROR_REASON_CODE_CODE_XPATH))
+            .sorted()
+            .collect(toList());
+    }
+
+    private boolean isCaseNotFound(List<String> errors) {
+        return CASE_NOT_FOUND_ERROR_CODES.equals(errors);
```

Comment:
Both codes must be present to identify response as "case not found"?
What if Spine returns "case not found" code but also some other additional? This check won't work then.

Maybe something like "errors contains CASE_NOT_FOUND_ERROR_CODES" would be more safe?

Reply:
Yes, exactly both codes.